feat: pub global binaries for most tools

xavierchanth opened the pull request:
<!--
Please make sure you've read and understood our contributing guidelines in CONTRIBUTING.md

If this is a bug fix, make sure your description includes "fixes #xxxx", or
"closes #xxxx"

Please provide the following information:
-->

**- What I did**

Fixed a couple uppercase variable names to lowercase.
i.e. "Name" -> "name"

Added binary definitions to the pubspecs for:

- at_cli
- at_pkam
- at_hive_recovery
- at_ve_doctor

I didn't do at_cram, because it looked like there was some more work to be done to cull it into shape and this was meant to be a quick fix.

**- How I did it**

**- How to verify it**
Pull branch

`dart pub global activate -s path <path to package>`

**- Description for the changelog**
feat: pub global binaries for most tools
<!--
Write a short (one line) summary that describes the changes in this
pull request for inclusion in the changelog:
-->

xavierchanth committed:
style: fix variable casing

xavierchanth committed:
feat: pub global binary for at_pkam

xavierchanth committed:
feat: pub global binary for at_cli

xavierchanth committed:
style: fix variable casing

xavierchanth committed:
feat: pub global binary for at_ve_doctor

xavierchanth committed:
fix: use correct file name for binary in at_cli

xavierchanth committed:
feat: pub global binary for at_hive_recovery

xavierchanth commented:
Once published, then anyone can do:
`pub global activate at_pkam`

and `at_pkam` is available as a command, exactly how `at_app` does it.

Same for all the other packages mentioned above.

cpswan commented on packages/at_cli/pubspec.yaml:
```diff
@@ -1,9 +1,12 @@
 name: at_cli
 description: A command line utility for at protocol commands.
-version: 3.1.0
+version: 3.1.0+1
```
Why 3.1.0+1 rather than 3.1.1 (or even 3.2.0 as this adds functionality rather than just patching)?

xavierchanth replied:
Nothing functionally in the package changed at all, the only purpose for republishing is to make the binary available through pub, hence why I considered it a rebuild.

xavierchanth replied:
Would this be considered a patch or a new build?

cpswan replied:
I'd consider the ability to `pub global` as added functionality even though it's maybe *of* the package rather than *in* the package.

cpswan commented on packages/at_cram/bin/at_cram.dart:
```diff
@@ -4,8 +4,8 @@ import 'package:crypto/crypto.dart';
-  var secret = getSecret(Name);
+  var name = arguments[0];
+  var secret = getSecret(name);
   secret = secret.trim();
```
could this just be `final secret = getSecret(arguments[0]).trim()`?

cpswan commented on packages/at_hive_recovery/pubspec.yaml:
```diff
@@ -3,6 +3,9 @@ description: A command-line utility to recover corrupt hive box
 version: 1.0.0
```
Bump this version too?

cpswan commented on packages/at_pkam/bin/at_pkam.dart:
```diff
@@ -6,8 +6,8 @@ import 'package:crypton/crypton.dart';
-  var privateKey = getSecret(Name);
+  var name = arguments[0];
+  var privateKey = getSecret(name);
   privateKey = privateKey.trim();
```
Sam as with secret for at_cram

cpswan commented on packages/at_pkam/pubspec.yaml:
```diff
@@ -1,8 +1,11 @@
 name: at_pkam
 description: A command-line utility to generate PKAM digest
-version: 1.1.0
+version: 1.1.0+1
```
again, correct use of semver?

cpswan commented on packages/at_ve_doctor/pubspec.yaml:
```diff
@@ -1,9 +1,12 @@
 description: A sample command-line application.
 publish_to: none
-version: 1.1.0
+version: 1.1.0+1
```
semver?

cpswan reviewed (commented):
Some nits before we press ahead.

Though I like the idea of doing this.

xavierchanth committed:
chore: fix versioning and update changelogs

xavierchanth commented:
Switched all of the versioning to be an increase in patch, rather than build.

Added missing changelog to at_hive_recovery.

Most of these packages are actually not published on pub.dev, so I will do that following this merge.

Also, switched the secret code to a oneliner as per the recommendation.
